### PR TITLE
[MINOR][SQL] Standardize 'continuous queries' to 'streaming Datasets/DataFrames'

### PR DESCRIPTION
**Pull request by lw-lin**
## What changes were proposed in this pull request?

This patch does some replacing (as `streaming Datasets/DataFrames` is the term we've chosen in [SPARK-15593](https://github.com/apache/spark/commit/00c310133df4f3893dd90d801168c2ab9841b102)):
- `continuous queries` -> `streaming Datasets/DataFrames`
- `non-continuous queries` -> `non-streaming Datasets/DataFrames`

This patch also adds `test("check foreach() can only be called on streaming Datasets/DataFrames")`.
## How was this patch tested?

N/A

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala [hunk 1]
```diff
@@ -433,8 +433,7 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
   @Experimental
   def foreach(writer: ForeachWriter[T]): ContinuousQuery = {
     assertNotBucketed("foreach")
-    assertStreaming(
-      "foreach() can only be called on streaming Datasets/DataFrames.")
+    assertStreaming("foreach() can only be called on streaming Datasets/DataFrames.")
```
Comment (lw-lin) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala]: these two lines are merged

File: sql/core/src/test/scala/org/apache/spark/sql/streaming/test/DataFrameReaderWriterSuite.scala [hunk 2]
```diff
@@ -371,66 +371,80 @@ class DataFrameReaderWriterSuite extends StreamTest with BeforeAndAfter {
...
+      override def open(partitionId: Long, version: Long): Boolean = false
+      override def process(value: String): Unit = {}
+      override def close(errorOrNull: Throwable): Unit = {}
+    }
+    val e = intercept[AnalysisException](w.foreach(foreachWriter))
+    Seq("foreach()", "streaming Datasets/DataFrames").foreach { s =>
+      assert(e.getMessage.toLowerCase.contains(s.toLowerCase))
+    }
+  }
+
```
Comment (lw-lin) [inline sql/core/src/test/scala/org/apache/spark/sql/streaming/test/DataFrameReaderWriterSuite.scala]: here we add a new test, checking foreach() can only be called on streaming Datasets/DataFrames

**Comment by lw-lin** [comment]
@tdas @zsxwing , would you mind taking a look? Thanks! :-)

**Comment by zsxwing** [comment]
@lw-lin thanks for the PR. However, we have not yet finalized the terms...

**Comment by tdas** [comment]
Yes. we are still reviewing the final API, so hold it for a bit. Thanks for noticing the inconsistencies.

**Comment by lw-lin** [comment]
@zsxwing @tdas, sure, this can wait. Thanks!

**Comment by rxin** [comment]
@tdas this matches what we discussed right?

**Comment by tdas** [comment]
Yes. This matches what we have discussed. So I am merging this to master and 2.0. Since I am working on a PR to change class names, I will do another pass to harmonized them.

**Comment by tdas** [comment]
Actually, I reverted this patch because it broke master build. And as I am refactoring based on the final API, I realize the refactoring is so large that PR is not really useful any more. So I wont be merging it again. Sorry about that.

**Comment by lw-lin** [comment]
Thanks! 
This patch introduced an compilation error because `DataFrameReader.text`'s return type had been changed back to `DataFrame` very recently, and I should have noticed this and updated this patch accordingly. Sorry for the inconvenience.

**Pull request closed without merging**